ecosystem/indexer-grpc/indexer-grpc-parser/src/worker.rs

```diff
@@ -251,30 +258,64 @@ impl Worker {
             let mut tasks = vec![];
             for transactions in transactions_batches {
                 let processor_clone = processor.clone();
+                let auth_token = self.config.indexer_grpc_auth_token.clone();
                 let task = tokio::spawn(async move {
                     let start_version = transactions.as_slice().first().unwrap().version;
                     let end_version = transactions.as_slice().last().unwrap().version;
+                    let txn_time_in_secs = transactions
+                        .as_slice()
+                        .first()
+                        .unwrap()
+                        .timestamp
+                        .as_ref()
+                        .map(|t| t.seconds as f64 + t.nanos as f64 * 1e-9);
+                    let current_time_in_secs = SystemTime::now()
+                        .duration_since(UNIX_EPOCH)
+                        .unwrap()
+                        .as_secs_f64();
 
-                    PROCESSOR_INVOCATIONS
+                    info!(
+                        request_token = auth_token,
+                        processor_name = processor_name,
+                        start_version = start_version,
+                        end_version = end_version,
+                        "[Parser] Received chunk of transactions."
+                    );
+                    if let Some(t) = txn_time_in_secs {
+                        PROCESSOR_DATA_RECEIVED_LATENCY_IN_SECS
+                            .with_label_values(&[auth_token.as_str(), processor_name])
+                            .set(current_time_in_secs - t);
```

Comment:
Could this use `time_diff_since_pb_timestamp_in_secs`?

Reply:
done.